### PR TITLE
Develop generate schedules filters

### PR DESCRIPTION
**Pull request by yzhao179**

### CODE REVIEW

**Review by jshen1s1 — CHANGES_REQUESTED** [review]
several commits behind the main branch 
merging will cause errors

File: frontend/src/components/GenerateSchedulesUnitAndTimePicker.js [hunk 1]
```diff
@@ -1,62 +1,184 @@
...
                     <Col>
                         <text>Minimum Units</text>
-                    <FormGroup controlId="integer">
+                    <form className={classes.root} noValidate autoComplete="off">
+                        <input
+                            type="text"
+                            id="outline-basic"
+                            placeholder="12-19"
+                            value={unit}
+                            onChange={setMax(this.target.value())}
```
Comment (colinaguan) [inline frontend/src/components/GenerateSchedulesUnitAndTimePicker.js]: (e) => setMax(e.target.value)

File: frontend/src/components/GenerateSchedulesUnitAndTimePicker.js [hunk 2]
```diff
@@ -1,62 +1,184 @@
...
                     <Col>
                         <text>Maximum Units</text>
-                    <FormGroup controlId="integer">
+                        <form className={classes.root} noValidate autoComplete="off">
+                            <input
+                                type="text"
+                                id="outline-basic"
+                                placeholder="12-19"
+                                value={unit}
+                                onChange={setMin(this.target.value())}
```
Comment (colinaguan) [inline frontend/src/components/GenerateSchedulesUnitAndTimePicker.js]: (e) => setMin(e.target.value)

File: frontend/src/components/CardSample.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
Comment (colinaguan) [inline frontend/src/components/CardSample.tsx:1]: if this file isn't needed, delete

File: frontend/src/components/ClassSearchFilters.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (colinaguan) [inline frontend/src/components/ClassSearchFilters.js:1]: not up to date

File: frontend/src/components/LoginPage.js [hunk 5]
```diff
@@ -1,49 +1,9 @@
-import React,{ useState } from 'react';
-import Login from "./Login"
-import Signup from "./Signup"
-import "../style/Login.css"
-import Alert from 'react-bootstrap/Alert';
+import React from 'react';
```
Comment (colinaguan) [inline frontend/src/components/LoginPage.js:1]: not up to date

File: frontend/src/components/GenerateSchedulesUnitAndTimePicker.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+import {Select} from "@material-ui/core";
+import InputLabel from '@material-ui/core/InputLabel';
+import { makeStyles } from '@material-ui/core/styles';
+import MenuItem from '@material-ui/core/MenuItem';
+import TextField from '@material-ui/core/TextField';
+import FormControl from '@material-ui/core/FormControl';
+import Container from '@material-ui/core/Container';
+
+import {Label} from "@material-ui/icons";
+const useStyles = makeStyles((theme) => ({
```
Comment (colinaguan) [inline frontend/src/components/GenerateSchedulesUnitAndTimePicker.js]: why not use a css file?

File: frontend/src/components/GenerateSchedulesUnitAndTimePicker.js [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+            minUnit=e;
+        }
+    }
+
+
+    const handleChange = (event) => {
+        setDate(event.target.value);
+    };
+    async function handleSubmit(){
+        if(maxUnit>19||minUnit>19||maxUnit<12||minUnit<12||maxUnit<minUnit)
```
Comment (colinaguan) [inline frontend/src/components/GenerateSchedulesUnitAndTimePicker.js]: units should be allowed to exceed 19

File: frontend/src/components/GenerateSchedulesUnitAndTimePicker.js [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+    },
+    textField: {
+        marginLeft: theme.spacing(1),
+        marginRight: theme.spacing(1),
+        width: 200,
+    },
+}));
+
+
+export default function GenerateSchedulesUnitAndTimePicker() {
```
Comment (colinaguan) [inline frontend/src/components/GenerateSchedulesUnitAndTimePicker.js]: (note for later reference) setting minimum units, maximum units, and time constraints will be passed as props

File: frontend/src/components/GenerateSchedulesUnitAndTimePicker.js [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+            </Modal>
+<Container fixed>
+
+<Row>
+    <Col>
+                <Row>
+
+                    <Col>
+                        <text>Minimum Units</text>
+                    <form className={classes.root} noValidate autoComplete="off">
```
Comment (colinaguan) [inline frontend/src/components/GenerateSchedulesUnitAndTimePicker.js]: if this is the bootstrap Form component, should be capitalized

File: frontend/src/components/GenerateSchedulesUnitAndTimePicker.js [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+                                }}
+                            />
+                        </form>
+                    </Col>
+
+                </Row>
+    </Col>
+    <Col>
+    <Button
+type="submit"
```
Comment (colinaguan) [inline frontend/src/components/GenerateSchedulesUnitAndTimePicker.js]: fix indentation

File: frontend/src/components/GenerateSchedulesUnitAndTimePicker.js [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+id="generatePageSearch"
+onClick={handleSubmit}
+    >Search</Button>
+    </Col>
+</Row>
+
+</Container>
+
+
+        </>
```
Comment (colinaguan) [inline frontend/src/components/GenerateSchedulesUnitAndTimePicker.js:96]: this outer element isn't needed if the whole thing is wrapped in a component (row or container)

File: frontend/src/components/GenerateSchedulesUnitAndTimePicker.js [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+                <Modal.Body>Please type in number between 12 and 19 again</Modal.Body>
+                <Modal.Footer>
+                    <Button variant="secondary" onClick={handleClose}>
+                        Close
+                    </Button>
+                </Modal.Footer>
+            </Modal>
+<Container fixed>
+
+<Row>
```
Comment (colinaguan) [inline frontend/src/components/GenerateSchedulesUnitAndTimePicker.js]: fix indentation

File: frontend/src/components/GenerateSchedulesUnitAndTimePicker.js [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+    <Col>
+                <Row>
```
Comment (colinaguan) [inline frontend/src/components/GenerateSchedulesUnitAndTimePicker.js]: The whole filters component should be a form, so you can use Form.Row and Form.Col elements. Having each filter element be its own form isn't the best solution. See the current ClassSearchFilters.js file for reference.

**Review by colinaguan — CHANGES_REQUESTED** [review]
The branch isn't up to date with the main; some file contents are completely replaced by old commits. If we merge this as is, it will be very problematic. Remember that what is in this branch should be a full working product off of what we already have. If the changes are too deep, just take your work, put it in a GenerateFilters.js file in local, make a new branch off of the updated main branch, and re-add your component to the GenerateSchedulesPage.js file.

Spacing and readability could be better.

Any and all warnings should be fixed (unused variables, etc.)

**Review by Brenden-Greenlee — APPROVED** [review]

**Comment by colinaguan** [comment]
filters work has been moved to another branch

**Pull request closed without merging**